durabletask/worker.py

```diff
@@ -353,44 +353,92 @@ def is_replaying(self) -> bool:
     def current_utc_datetime(self, value: datetime):
         self._current_utc_datetime = value
 
-    def create_timer(self, fire_at: Union[datetime, timedelta]) -> task.Task:
+    def create_timer(self, fire_at: Union[datetime, timedelta],
+                     retryable_task: Optional[task.RetryableTask] = None,
+                     retryable_type: Optional[str] = None) -> task.Task:
         id = self.next_sequence_number()
         if isinstance(fire_at, timedelta):
             fire_at = self.current_utc_datetime + fire_at
         action = ph.new_create_timer_action(id, fire_at)
         self._pending_actions[id] = action
 
-        timer_task = task.CompletableTask()
+        timer_task = task.TimerTask()
+        if retryable_task is not None and retryable_type is not None:
+            timer_task.set_retryable_parent(retryable_task, retryable_type)
         self._pending_tasks[id] = timer_task
         return timer_task
 
     def call_activity(self, activity: Union[task.Activity[TInput, TOutput], str], *,
-                      input: Union[TInput, None] = None) -> task.Task[TOutput]:
+                      input: Optional[TInput] = None,
+                      retry_policy: Optional[task.RetryPolicy] = None) -> task.Task[TOutput]:
         id = self.next_sequence_number()
-        name = activity if isinstance(activity, str) else task.get_name(activity)
-        encoded_input = shared.to_json(input) if input else None
-        action = ph.new_schedule_task_action(id, name, encoded_input)
-        self._pending_actions[id] = action
-
-        activity_task = task.CompletableTask[TOutput]()
-        self._pending_tasks[id] = activity_task
-        return activity_task
+        
+        self.call_activity_util(id, activity, input, retry_policy)
+        return self._pending_tasks.get(id, task.CompletableTask())
+
+    def call_activity_util(self, id: Optional[int],
+                           activity: Union[task.Activity[TInput, TOutput], str],
+                           input: Optional[TInput] = None,
+                           retry_policy: Optional[task.RetryPolicy] = None,
+                           act_task: Optional[task.CompletableTask[TOutput]] = None):
+            if id is None:
+                id = self.next_sequence_number()
+            name = activity if isinstance(activity, str) else task.get_name(activity)
+                    
+            if act_task is None:
+                encoded_input = shared.to_json(input) if input else None
+            else:
+                encoded_input = input.__str__()
```

Comment:
Why do we do `__str__()` here? Can you add a comment explaining this?

Reply:
Added comment:

```
# Here, we don't need to convert the input to JSON because it is already converted.
# We just need to take string representation of it.
```

Reply:
`input` here is of type `TInput@call_activity_function_helper`.
And, below `new_schedule_task_action` expects `encoded_input` to be of type `str`.
I have removed `__str__()` but rather typecasted to `str` now.